python/pyspark/worker.py

```diff
@@ -357,8 +357,14 @@ def map_batch(batch):
             num_output_rows = 0
             for result_batch, result_type in result_iter:
                 num_output_rows += len(result_batch)
-                assert is_map_iter or num_output_rows <= num_input_rows[0], \
-                    "Pandas MAP_ITER UDF outputted more rows than input rows."
+
+                if is_scalar_iter and num_output_rows != num_input_rows[0]:
```

Comment:
@WeichenXu123, I think we should either keep this condition or remove.
Currently, it requires each series in the iterator to be the same size - it could fail fast.
If we remove this, it will require each iterator to be the same size - it will require to compute the whole iterator.

I remember you used this for ML side, right? does both satisfy your case, and general ML cases?

Reply:
each series in the iterator to be the same size - it could fail fast. Keep this will be sufficient for  ML cases.